src/compiler/checker.ts

```diff
@@ -30099,18 +30099,27 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             isTypeAssignableToKind(checkComputedPropertyName(firstDecl.name), TypeFlags.ESSymbol));
     }
 
-    function getObjectLiteralIndexInfo(node: ObjectLiteralExpression, offset: number, properties: Symbol[], keyType: Type): IndexInfo {
+    function getObjectLiteralIndexInfo(offset: number, properties: Symbol[], keyType: Type, isReadonly: boolean): IndexInfo {
         const propTypes: Type[] = [];
         for (let i = offset; i < properties.length; i++) {
             const prop = properties[i];
-            if (keyType === stringType && !isSymbolWithSymbolName(prop) ||
-                keyType === numberType && isSymbolWithNumericName(prop) ||
-                keyType === esSymbolType && isSymbolWithSymbolName(prop)) {
+            if (keyType === numberType && isSymbolWithNumericName(prop) || keyType === esSymbolType && isSymbolWithSymbolName(prop)) {
                 propTypes.push(getTypeOfSymbol(properties[i]));
             }
+            else if (!isSymbolWithSymbolName(prop)) {
+                if (keyType === stringType) {
+                    propTypes.push(getTypeOfSymbol(properties[i]));
+                }
+                else {
+                    const source = tryCast(prop, isTransientSymbol)?.links.computedNameType || getStringLiteralType(unescapeLeadingUnderscores(prop.escapedName));

```

Comment:
This line is finicky but I wasn't sure what's the proper way of accessing a symbol link (likely I should change this to `getSymbolLinks(prop)`?) and what transient symbols are or when they should be used.